tests/phpunit/tests/post/revisions.php

```diff
@@ -866,4 +866,115 @@ public function test_wp_get_post_revisions_url_returns_null_with_revisions_disab
 
 		add_post_type_support( 'post', 'revisions' );
 	}
+
+	/**
+	 * Tests that wp_save_post_revision() respects the 'wp_save_post_revision_revisions_before_deletion' filter
+	 * when deleting revisions.
+	 *
+	 * This test should protect the original revision, send the rest to be checked against wp_revisions_to_keep(),
+	 * and result in two revisions: The latest revision, and the original.
+	 *
+	 * @ticket 57320
+	 *
+	 * @covers ::wp_save_post_revision
+	 */
+	public function test_wp_save_post_revision_should_respect_revisions_before_deletion_filter() {
+		$post_id = self::factory()->post->create( array( 'post_title' => 'Test 57320' ) );
+
+		add_filter(
+			'wp_revisions_to_keep',
+			static function() {
+				return 1;
+			}
+		);
+
+		add_filter(
+			'wp_save_post_revision_revisions_before_deletion',
+			static function ( $revisions ) {
+				// Ignore the first revision and return the rest for deletion.
+				return array_slice( $revisions, 1 );
+			}
+		);
+
+		for ( $update = 1; $update < 4; ++$update ) {
+			wp_update_post(
+				array(
+					'ID'         => $post_id,
+					'post_title' => 'Test 57320 Update ' . $update,
+				)
+			);
+		}
+
+		$actual = wp_get_post_revisions( $post_id );
+
+		$this->assertCount(
+			2,
+			$actual,
+			'There should be two revisions.'
+		);
```

Comment:
```suggestion
		$actual = wp_get_post_revisions( $post_id );
		$total_revisions = count( $actual );

		$this->assertCount(
			2,
			$total_revisions,
			'There should be two revisions.'
		);
```

`wp_get_post_revisions` returns array as per [return document](https://developer.wordpress.org/reference/functions/wp_get_post_revisions/#return). @costdev is the change make sense?

Reply:
[assertCount()](https://phpunit.readthedocs.io/en/9.5/assertions.html#assertcount) counts the elements in its `$actual` parameter, so the assertion is used correctly in the test.

Reply:
Make sense. Thanks